src/code/z_message_PAL.c

```diff
@@ -1873,14 +1873,14 @@ void Message_StartOcarina(PlayState* play, u16 ocarinaActionId) {
     }
 }
 
-void func_8010BD58(PlayState* play, u16 ocarinaActionId) {
-    play->msgCtx.unk_E40E = 0;
-    Message_StartOcarina(play, ocarinaActionId);
+void Message_StartOcarina(PlayState* play, u16 ocarinaActionId) {
+    play->msgCtx.disableSunsSong = false;
+    Message_StartOcarinaImpl(play, ocarinaActionId);
 }
 
-void func_8010BD88(PlayState* play, u16 ocarinaActionId) {
-    play->msgCtx.unk_E40E = 1;
-    Message_StartOcarina(play, ocarinaActionId);
+void Message_StartOcarinaSunsSongDisabled(PlayState* play, u16 ocarinaActionId) {
+    play->msgCtx.disableSunsSong = true;
+    Message_StartOcarinaImpl(play, ocarinaActionId);
```

Comment:
`SunsSongDisabled` makes me think that it won't register suns song as all, instead of not applying the time-forward effect. For example, this is called by Darunia, so you would expect Suns to not be a valid song to play. However, playing Suns will trigger Darunia to dance to the "wrong song" (see `gGoronCityDaruniaWrongCs`), so that seems to contradict suns being disabled.

What about `disableSunsSongEffect` or `blockSunsSongEffect`?

Reply:
I did think of this, and had it named after the "time skip" aspect at first.
But really, all it does is stop the suns song state from being set in the ocarina effect when it destroys, `disableSunsSongEffect` is okay, other than the age old problem of "dont call things effects cause it might sound like a particle effect" lol
but, it may not be too much of a problem

Reply:
also should be noted I took inspiration for this name from the instance variable it lives right next to, which has the same issue (you can still play warps songs, just cant warp)

```
   /* 0xE40C */ s16    disableWarpSongs; // warp song flag set by scene commands
    /* 0xE40E */ s16    disableSunsSong;
```
scene restrictions are the same, the song is still playable but its still referred to in this way
we can try and fix this problem everywhere, but I also kinda think its fair to let the reader figure out that it means the result of the song and not the ability to play it at all

Reply:
I see, the parallel between `disableWarpSongs` and `disableSunsSong` does make sense. Maybe a comment on:
`/* 0xE40E */ s16    disableSunsSong; // comment`
would help